pkg/scheduler/framework/plugins/noderesources/balanced_allocation_test.go

```diff
@@ -269,6 +276,24 @@ func TestNodeResourcesBalancedAllocation(t *testing.T) {
 				{Spec: machine2Spec, ObjectMeta: metav1.ObjectMeta{Labels: labels1}},
 			},
 		},
+		{
+			// Node1 scores on 0-MaxNodeScore scale
+			// CPU Fraction: 300 / 250 = 100%
+			// Memory Fraction: 600 / 10000 = 60%
+			// Node1 Score: MaxNodeScore - (100-60)*MaxNodeScore = 60
+			// Node2 scores on 0-MaxNodeScore scale
+			// CPU Fraction: 100 / 250 = 40%
+			// Memory Fraction: 200 / 10000 = 20%
+			// Node2 Score: MaxNodeScore - (40-20)*MaxNodeScore= 80
+			pod:          &v1.Pod{Spec: nonZeroContainer},
+			nodes:        []*v1.Node{makeNode("machine1", 250, 1000*1024*1024), makeNode("machine2", 250, 1000*1024*1024)},
+			expectedList: []framework.NodeScore{{Name: "machine1", Score: 60}, {Name: "machine2", Score: 80}},
+			name:         "no resources requested, pods scheduled",
```

Comment:
originally, it will return `"machine1": 0, "machine2": 80`, machine2 is preferred for both cases, consider that there are other types of scoring plugin, the pod should be scheduled in anyway.

Reply:
Correct, you have to think of other plugins.
If we score 0 vs 80, then this score has a big difference, and takes prevalence over other plugins. 60 vs 80 only gives a difference of 20, then other more important scores can be taken into account.